### PR TITLE
fix: Loading state changes button width 

### LINKED ISSUE #25890: Fix button width for loading state
When setting `isLoading` flag the width of the button should not change and should be according to the content.

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #25890

### CODE REVIEW

File: app/client/packages/design-system/widgets/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -53,24 +53,22 @@ export const Button = forwardRef(
     const { visuallyHiddenProps } = useVisuallyHidden();
 
     const renderChildren = () => {
-      if (isLoading) {
-        return (
-          <>
+      return (
```
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/components/Button/Button.tsx:56]: We were previously conditionally rendering loader or content. But now we are rendering both. We are hiding loader based on isLoading state with CSS

File: app/client/packages/design-system/widgets/src/components/Button/index.styled.tsx [hunk 2]
```diff
@@ -67,51 +67,64 @@ export const buttonStyles = css<StyledButtonProps>`
...
-  block-size: var(--sizing-8);
   border-radius: var(--border-radius-1);
   user-select: none;
-  min-inline-size: var(--sizing-8);
   position: relative;
   font-weight: 600;
   border-width: 0;
   border-style: solid;
   font-family: inherit;
+  overflow: hidden;
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/Button/index.styled.tsx:79]: Off topic. But do we need to add here or some other values fro padding to rewrite user agent styles? `padding: 0;` ?

Снимок экрана 2023-08-02 в 09 58 24

cc @ichik

File: app/client/packages/design-system/widgets/src/components/Button/Button.tsx [hunk 3]
```diff
@@ -53,24 +53,22 @@ export const Button = forwardRef(
...
+            <Text lineClamp={1} textAlign="center">
+              {children}
+            </Text>
+          </span>
+
+          <span aria-hidden={!isLoading ? true : undefined} data-loader="">
             <HeadlessIcon>
               <Spinner />
             </HeadlessIcon>
             {/* TODO(pawan): How to make sure "Loading..." text is internationalized? */}
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Button/Button.tsx:69]: I've been looking at this for a long time, let's just add prop for this with a default value `Loading...`. Internationalization should be handled at the app level anyway.

File: app/client/packages/design-system/widgets/src/components/Button/index.styled.tsx [hunk 4]
```diff
@@ -133,14 +146,29 @@ export const StyledButton = styled(HeadlessButton)<StyledButtonProps>`
...
     cursor: default;
     /** adding opacity 1 here because we are lowering opacity for aria-disabled and when loading is true, aria-disabled is also true  */
     opacity: 1;
   }
+
+  & [data-loader] {
+    display: none;
+  }
+
+  &[data-loading] [data-loader] {
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Button/index.styled.tsx:163]: Nice 👍

File: app/client/packages/design-system/widgets/src/components/Button/index.styled.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ichik) [reply app/client/packages/design-system/widgets/src/components/Button/index.styled.tsx]: So we don't do any css reset?

[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/Button/index.styled.tsx]: Nope and as far as I know, we shouldn't. I wouldn't want to have some kind of dependency on this, so we can fix these things ourselves by adding a few properties.

[Here](https://css-tricks.com/normalize-css-no/) some old article about this.

[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/Button/index.styled.tsx]: I got you, but I'm sure that for our simple components, such as input and button, we can apply the reset styles on our side, without any dependencies. Anyway, we have to test our components in different browsers and if something goes wrong, we will fix it. So I would just set pudding for our button, and not use the default one, and that's it.

**Review by KelvinOm — APPROVED** [review]

**Review by KelvinOm — APPROVED** [review]

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**